core/src/test/java/com/linecorp/armeria/client/RefreshingAddressResolverTest.java

```diff
@@ -444,6 +447,74 @@ void shouldRefreshWhenDnsCacheIsExpired() {
         }
     }
 
+    @Test
+    void shouldRefreshWhenSearchDnsCacheIsExpired() throws InterruptedException {
+        try (TestDnsServer server = new TestDnsServer(ImmutableMap.of(
+                new DefaultDnsQuestion("foo.com.", A),
+                new DefaultDnsResponse(0).addRecord(ANSWER, newAddressRecord("foo.com.", "1.1.1.1")),
```

Comment:
Should we add a test for `DnsQuestionWithoutTrailingDot.of("foo.com.", A)` in this method, just in case it's resolved to a wrong IP address (i.e. not `1.1.1.1`)?

Reply:
Added with [`4f5470e` (#4133)](https://github.com/line/armeria/pull/4133/commits/4f5470e80f4b17cdfce181a5acd0ce919708b5d5)